src/flexmock/api.py

```diff
@@ -311,75 +313,62 @@ def updated(self: Any) -> Any:
-                            raise (
-                                ExceptionMessageError(
-                                    'expected /%s/, raised "%s"'
-                                    % (args["kargs"][0].pattern, message)
-                                )
-                            )
-                    elif expected_message and expected_message != message:
+            return_values = expectation._return_values
+            if not return_values:
+                raise
+            raised, instance = sys.exc_info()[:2]
+            assert raised, "no exception was raised"
+            message = "%s" % instance
+            expected = return_values[0].raises
+            if not expected:
+                raise
+            args = return_values[0].value
+            expected_message = "%s" % expected(*args["kargs"], **args["kwargs"])
+            if inspect.isclass(expected):
+                if expected is not raised and expected not in raised.__bases__:
+                    raise ExceptionClassError("expected %s, raised %s" % (expected, raised))
+                if args["kargs"] and isinstance(args["kargs"][0], RE_TYPE):
+                    if not args["kargs"][0].search(message):
                         raise (
                             ExceptionMessageError(
-                                'expected "%s", raised "%s"' % (expected_message, message)
+                                'expected /%s/, raised "%s"' % (args["kargs"][0].pattern, message)
                             )
                         )
-                elif expected is not raised:
-                    raise ExceptionClassError('expected "%s", raised "%s"' % (expected, raised))
-            else:
-                raise
+                elif expected_message and expected_message != message:
+                    raise (
+                        ExceptionMessageError(
+                            'expected "%s", raised "%s"' % (expected_message, message)
+                        )
+                    )
+            elif expected is not raised:
+                raise ExceptionClassError('expected "%s", raised "%s"' % (expected, raised))
 
         def match_return_values(expected: Any, received: Any) -> bool:
-            if not isinstance(expected, tuple):
-                expected = (expected,)
-            if not isinstance(received, tuple):
-                received = (received,)
-            if len(received) != len(expected):
-                return False
-            for i, val in enumerate(received):
-                if not _arguments_match(val, expected[i]):
-                    return False
-            return True
+            expected = (expected,) if not isinstance(expected, tuple) else expected
+            received = (received,) if not isinstance(received, tuple) else received
+            return len(received) == len(expected) and all(
+                _arguments_match(r, e) for r, e in zip(received, expected)
+            )
```

Comment:
We can be smart with itertools also:
```suggestion
            return all(
                _arguments_match(r, e) for r, e in itertools.zip_longest(received, expected, fillvalue=object())
            )
```
(changing the fillvalue to a new object ensures that nothing will be equal to it)

I'm not saying we need to change it, it might also be considered too smart and harder to maintain, but I find it interesting :smile: